benchopt/utils/safe_import.py

```diff
@@ -1,31 +1,87 @@
 import sys
 import warnings
 import importlib
+import importlib.util
+
+from importlib.abc import Loader, MetaPathFinder
 from pathlib import Path
+from unittest.mock import Mock
 
 from joblib.externals import cloudpickle
 
 from ..config import RAISE_INSTALL_ERROR
 
-SKIP_IMPORT = False
+
+MOCK_ALL_IMPORT = False
 BENCHMARK_DIR = None
 PACKAGE_NAME = "benchmark_utils"
 
 
-class SkipWithBlock(Exception):
-    pass
+class MockLoader(Loader):
+    def __init__(self, name):
+        self.name = name
 
+    def create_module(self, spec):
+        return Mock(name=self.name)
```

Comment:
I don't know if it would be worth it, but it is possible to configure `Mock` sub-classes to throw errors when some event happens. For example, when an attribute is called (`mocked_module.attribute()` raises an exception). With the base `Mock` object, calls return another `Mock` object.

For more versatility, we could also use [`MagicMock`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.MagicMock) which implements `__getitem__` and `__setitem__`. This would enable things like `mocked_module.CONFIGS["key"]`.